topic_dynamics/parsing.py

```diff
@@ -0,0 +1,376 @@
+        for date in tqdm(range(2, len(dates) + 1)):
+            starting_index_diff = count_index_diff + 1
+            # Save the tokens of the "previous" slice into memory
+            previous_version = {}
+            with open(os.path.abspath(os.path.join(slices_tokens_dir, str(date - 1) + ".txt"))) as fin:
+                for line in fin:
+                    previous_version[line.rstrip().split(";")[1]] = line.rstrip().split(";")[2]
+            current_version = []
+            with open(os.path.abspath(os.path.join(slices_tokens_dir, str(date) + ".txt"))) as fin:
+                for line in fin:
+                    # Iterate over files in the "current" version
+                    current_version.append(line.rstrip().split(";")[1])
+                    address = line.rstrip().split(";")[1]
+                    tokens = {}
+                    for token in line.rstrip().split(";")[2].split(","):
+                        tokens[token.split(":")[0]] = int(token.split(":")[1])
+                    tokens = Counter(tokens)
+                    old_address = address.replace(dates[date - 1].strftime("%Y-%m-%d"),
+                                                  dates[date - 2].strftime("%Y-%m-%d"), 1)
+                    # Check if the file with this name existed in the previous version
+                    if old_address in previous_version.keys():
+                        old_tokens = {}
+                        for token in previous_version[old_address].split(","):
+                            old_tokens[token.split(":")[0]] = int(token.split(":")[1])
+                        old_tokens = Counter(old_tokens)
+                        # Calcualate which tokens have been added and removed between versions
+                        created_tokens = sorted((tokens - old_tokens).items(), key=itemgetter(1), reverse=True)
+                        deleted_tokens = sorted((old_tokens - tokens).items(), key=itemgetter(1), reverse=True)
+                        new_tokens = []
+                        if len(created_tokens) != 0:
+                            for token in created_tokens:
+                                new_token = "+" + token[0]
+                                new_tokens.append([new_token, token[1]])
+                        if len(deleted_tokens) != 0:
+                            for token in deleted_tokens:
+                                new_token = "-" + token[0]
+                                new_tokens.append([new_token, token[1]])
+                    # If the file is new, all of its tokens are considered created
+                    else:
+                        tokens = sorted(tokens.items(), key=itemgetter(1), reverse=True)
+                        new_tokens = []
+                        for token in tokens:
+                            new_token = "+" + token[0]
+                            new_tokens.append([new_token, token[1]])
```

Comment:
create separate function to convert `token_cnt` into `diff_token_count` - it's duplicated here and above